### ISSUE TITLE
Poor performance on aishell3 with gst+xvector+tacotron2 configuration

### ISSUE THREAD

**Issue by keshawnhsieh** [espnet/espnet#5126]
I ran aishell recipe with this [gst + xvector +tacotron2](https://github.com/espnet/espnet/blob/master/egs2/aishell3/tts1/conf/tuning/train_gst%2Bxvector_tacotron2.yaml) configuration. However the cloning voice is really poor both on dev set and test set. Specifically, the audio quality is much poorer than a standard TTS system and the alignment seems also has big problem, the synthesized audio sometimes miss the last few words in the given text.

I did some small modification to config file in order to use full memory of GPU or other purpose and keep the other setting not changed.

For tacotron2:
1. use speechrain to extract instead of Kaldi
add `--xvector_tool speechbrain` in run.sh below
https://github.com/espnet/espnet/blob/079efdb435ba01cd8b8b6a796cd9c8e29be61255/egs2/aishell3/tts1/run.sh#L38-L55
2. some minor changes in gst+xvector+tacotron2 [config file ](https://github.com/espnet/espnet/blob/master/egs2/aishell3/tts1/conf/tuning/train_gst%2Bxvector_tacotron2.yaml)
change speaker embedding dimentions to match the speechbtrain setup `spk_embed_dim: 192`
change batch_bins to 3 times largger to full utilize the 32G memory of V100 `batch_bins: 11250000`

I trained a parallel wavegan with [ParallelWaveGan](https://github.com/kan-bayashi/ParallelWaveGAN) on aishell3 train set at the same time.
The recipe I refer to is [parallel_wavegan.v1.yaml](https://github.com/kan-bayashi/ParallelWaveGAN/blob/master/egs/csmsc/voc1/conf/parallel_wavegan.v1.yaml) in csmsc recipe. Some minor changes also made just to fully ultilize the memory of GPU.
1. increase batch_size  from defaul 5 to 30
2. decrease discriminator_train_start_steps from 100000 to 20000
3. decrease train_max_steps from 400000 to 80000

The attention plots and loss curve of tacotron2 seems normal and spectral_convergence_loss of vocoder also looks good. I pin them here for reference.
gst+xvector+tacotron2 train/val loss:
![image](https://user-images.githubusercontent.com/29997158/232662242-adaebf52-dce6-4bf6-a62f-bf0fff6a46dd.png)
gst+vector+tacotron2 attention plot:
![image](https://user-images.githubusercontent.com/29997158/232662885-3212b939-85c1-4e84-b606-526d4536768c.png)
![image](https://user-images.githubusercontent.com/29997158/232662911-0e81ecb3-6b1d-4c47-a2e9-2d1393aba04f.png)
![image](https://user-images.githubusercontent.com/29997158/232662929-265e112a-8934-4737-bb73-a2893aaaca45.png)
parallel_wavegan spectral_convergence_loss:
![image](https://user-images.githubusercontent.com/29997158/232663044-ac4761dd-a853-49d6-ac63-209e29de61de.png)

I test the result with ep170 ckpt(as I change the batch bins 3 times larger than default setting, I think there is no need to train all 500 epochs)  and parallel_wavegan stopped at 80000 steps. But got some disappoint results. I attach first 4 audio samples in dev set and test set below.

dev set:
1. Most of synthesized audios sounds fuzzy compared to the groundtruth like case SSB00090031 and SSB00090289
2. And for SSB00110026 and SSB00110284, the sythesized audios miss the last few chinese word in the input text.
SSB00110026 only contains "人民法院在审理证券行政处罚" while the input shown in text file is "人民法院在审理证券行政处罚案件时".
SSB00110284 only contains "各地区各部门普遍以整治收送" while the input shown in text file is "各地区各部门普遍以整治收送礼金歪风为切入点".
[dev_set.zip](https://github.com/espnet/espnet/files/11258327/dev_set.zip)

test set:
The problem happened in test set is exactly the same as dev set.
1. All synthesized audio has kind of fuzzy feelings
2. SSB00050353 miss the last few word in input text.
[test_set.zip](https://github.com/espnet/espnet/files/11258483/test_set.zip)

**Comment by keshawnhsieh** [comment]
Thank you for your detailed reply on this, especially the guidance on how to locate the problem of vocoder part.
I ran the decode step of parallel wavegan and indeed found that the quality of synthesized audio was not good compare to a well trained csmsc model. I attached some audios for reference below.
[parallel_wavegan_samples.zip](https://github.com/espnet/espnet/files/11303292/parallel_wavegan_samples.zip)

And according to your advice, I started a new training task with default configuration. But I have a doubt if it's possible to train a multi-speaker vocoder to support both female and male. I looked into aishell3 and found it contains more female speakers(176) than male speakers(42). Do you have any opinion on this?

As for tacotron2 part, you didn't mention much on how to locate the problem. I further looked into some specific case and started a new training with totally default configuration. I want to state some new findings here.
Firstly, I checked the attention map after running the inference on test set. I found there are lots of pictures show that output only attends part of input and miss the last few words in the input. You can see it obvious in pictures shown below.
![SSB00800293](https://user-images.githubusercontent.com/29997158/233831350-e5a9eeca-ceff-4b3e-8648-ddaf76fb3a6f.png)
![SSB01120044](https://user-images.githubusercontent.com/29997158/233831392-ae860b70-95b1-4617-8e81-5367e03c5006.png)
![SSB01490039](https://user-images.githubusercontent.com/29997158/233831403-b28ea637-c91c-4acd-a8c3-eb4b5faad3f3.png)

Secondly, I found the new training I started with default configurations behaves strangely when I checked the attention map shown in tensorboard. The hot mapping line is one point apart from another like this:
![tts dec att 103ep](https://user-images.githubusercontent.com/29997158/233831638-d965a5e2-3274-42a0-b4d2-62c8f5ccdaec.png)
![tts dec att 103ep](https://user-images.githubusercontent.com/29997158/233831652-375f27b9-4f7a-483e-8ff1-31c89d813c99.png)
Although the training only finishs 103/500, but it seems not to be a good phenomenon. 

Do you have any suggestions on text2mel model part? Thanks in advance.
